### DISCUSSION TITLE
Cannot build protocol_examples_common

### DISCUSSION THREAD

**Discussion by swimleftproducts** [Mair/esp32-course#70] [Course Qeustions] [answered]
Hello All,

I am just starting the wifi section. The example connection video has us look at the protocol_examples_common example project. We then use this code in our project. However, when I tried to build the protocol_examples_common file I received this error ;
_CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as
    cmake_minimum_required(VERSION 3.20)_

Then after adding the cmake_minimum_required(VERSION 3.20)_ code I get the following error:
  _Unknown CMake command "idf_component_register"._

I while the google html downloading application worked that we built in this video, it still looks I am getting this error in the background. Looking around on the internet, it seems that this error could be due to using the wrong IDF version. However, I believe I am using the most up-to-date version.

I would appreciate any help on figuring out how to fix this error.

**Comment by Mair** [comment]
hi @swimleftproducts 
Is it maybe that you updated the wrong CMakeLists.txt file?
There are 2 in every project. One in the root and on in the main folder. You can review the code in this repo https://github.com/Mair/esp32-course/tree/master/_13_Internet_Connection/_13_1_exampleConnect

**Comment by swimleftproducts** [comment, marked as answer]
I ended up watching this video [components](https://www.youtube.com/watch?v=COB_vBtTdiY&t=2s) and it really helped me get up to speed with the cmake tool. 

Mair, hopefully you are ok with me posting this here, if not I will remove.

**Reply by swimleftproducts** [reply]
Also, to follow up a bit, I think the example component is set up to be only a component, not a project. As a result, it does not make the required cmake file language needed to kick off the cmake process. After building my own components I can sort of follow what is going on here. 

Hope this helps someone
